### ISSUE TITLE
Dialog - prevent undesired fullscreen-snapping on small viewports

### ISSUE THREAD

**Issue by ashetland** [Esri/calcite-design-system#6191]
**monday.com sync:** #18005399996

## Current behavior
Currently, when the viewport is small enough, a Dialog will snap to fullscreen **without** `fullscreen` being set. This behavior may be undesired and should be something the developer / designer should be able to set up themselves.

## Proposed changes
The following proposal would give the developer and designer more control over the component's fullscreen behaviors. 
1. Remove the automatic snapping to fullscreen behavior.
2. Setting `fullscreen` would always make a `dialog` fullscreen.

### Acceptance Criteria

1. Automatic behaviors removed
2. `fullscreen` prop always makes `dialog` fullscreen

### Relevant Info

cc @macandcheese, @asangma, @mitc7862, @paulcpederson for thoughts and opinions. 

### Which Component

Dialog

### Esri team

Calcite (design)

**Comment by macandcheese** [comment]
Hit close instead of Comment, 😬 whoops.

I’d like to get these in for 1.0 if possible. I’d like to hopefully clarify some of the above:

Fullscreen, when true, always makes a modal fullscreen - should css var for modal height and width override this? 

It is proposed that the “snap to full” is removed except when opted into. Should this be a prop, with the same “default widths” snapping to fullscreen as current, as well as support a separate custom width to do this?

“Snapping to full” currently makes the buttons full width when that change occurs. Does this need to be a separate opt-in? “Always fullscreen” will still need a px width at which this swap occurs.

How valuable is having the default s/m/l width with proposed css variables for width / height? Do we need these or a single default with customizability cover those cases?

Please advise on above from a design perspective :) @ashetland @SkyeSeitz

**Comment by jcfranco** [comment]
@ashetland @macandcheese The referenced PR above is already merged. Is there any pending for this or can it be marked as installed?

**Comment by macandcheese** [comment]
No, these would all be additive or breaking changes to make for future enhancements, nothing immediate.

**Comment by geospatialem** [comment]
This may have `visual changes` and/or `breaking changes` implications, adding a `spike` to determine next steps.

**Comment by geospatialem** [comment]
Needs re-evaluation/expertise from design with the `dialog` component addition in 2024, is this still valid, and if so, what is the recommended approach?

**Comment by macandcheese** [comment]
I think it’s valid, ideally “cover” or “fullscreen” behavior is always opted into by a user and doesn’t occur automatically, which can be unexpected.

Existing widths should probably respond by ensuring a small margin is always present around Dialog, when the viewport is smaller than the content. Apps can opt-in to the cover behavior as needed.

**Comment by SkyeSeitz** [comment]
Dialog MVP simply matched the behavior of Modal, but I agree - this is still a needed change. The current margin is 32px which can continue to be used instead of automatically snapping to cover.

**Comment by rebollin** [comment]
Commenting for visibility as I hit the issue described below:

>I've run into a bit of an odd calcite issue that not many others are able to reproduce. The dialog is opening full screen for me on my windows precision 5560 laptop. For example, here is what I am seeing on Firefox and chrome:
> ![Image](https://github.com/user-attachments/assets/b6dc5fcc-30f1-4116-bd6e-660de9f7b157)
> and this is what others see
> 
> ![Image](https://github.com/user-attachments/assets/b17ab651-ff39-4669-b08c-91b44dada82b)
> https://codepen.io/ptmdmusique/pen/raBQGJb?editors=1000 are others able to reproduce this at all? If I adjust the zoom on my browser to less that 100% it reduces the dialog size as expected. But the defaults didn't used to do this for me, so I found it odd 
> 

@macandcheese Commented that this is expected but could be improved via this issue 😄

**Comment by jcfranco** [comment]
@macandcheese @ashetland Do we have any users who rely on the current behavior without setting `fullscreen`? This change would be breaking for them. If it’s been causing more confusion than helping, we might want to treat it as a bug.

> User-definable viewport width for snapping to fullscreen, when desired

Do any of our components support something like this? If not, I’d prefer that we recommend using `ResizeObserver` to toggle `placement` accordingly.

**Comment by macandcheese** [comment]
Yeah, I'd agree we can leave that up to the implementer to determine when to adapt and change props / values. I had that "hackathon" Responsive Overrides idea that would ostensibly solve for this .. but .. not directly related here. 

I think we could proceed with treating as a bug.

**Comment by jcfranco** [comment]
Updated issue based on ☝️. LMK if any tweaks are needed.

**Comment by ashetland** [comment]
> Updated issue based on ☝️. LMK if any tweaks are needed.

LGTM!

**Comment by macandcheese** [comment]
Just a note - for Dialog - we have `placement="cover"` as the way to set this by a user.

**Comment by paulcpederson** [comment]
we rely on implicit snapping to fullscreen on mobile for all of our hundreds of dialogs in Online. Is the expectation that after this change we'd need to go through and update every single one with a prop on only certain screen sizes?

**Comment by jcfranco** [comment]
It would, yes. We could also provide some utils to make this simpler to set up. 

Curious, has the implicit behavior worked reliably out-of-the-box for your use case or have you had to make any adjustments?

**Comment by jcfranco** [comment]
Also, based on the above use case, I think we should consider this breaking.

**Comment by paulcpederson** [comment]
> Curious, has the implicit behavior worked reliably out-of-the-box for your use case or have you had to make any adjustments?

The basic behavior has been precisely what we want on every dialog. We have had to respond to some changes in how the footer actions were implemented but the basic behavior of fullscreen under the minimum size has worked very well.

**Comment by ashetland** [comment]
Agreed that this should probably be considered breaking. 

> The basic behavior has been precisely what we want on every dialog.

Are all these dialogs in the `modal` state? If so, maybe it makes sense to maintain this behavior when `modal="true"`, but change it for non-modal Dialogs? If this remains entirely unchanged, it wouldn't be possible to make a "small dialog in the middle of the screen (or other placement)" with space around it.

**Comment by paulcpederson** [comment]
Yeah all have `modal=true`. I actually haven't used a dialog without `modal` yet.

**Comment by jcfranco** [comment]
Belated update: we’ll be adding an opt-out property to mitigate the breaking change. We’ll revisit removal of this feature once a more flexible utility for configuring components per screen size is available ([#12496](https://github.com/Esri/calcite-design-system/issues/12496)￼).

**Comment by dmartini-zrh** [comment]
This is a very appreciated enhancement, looking forward to using it    👍

**Comment by geospatialem** [comment]
Verified with the new `fullscreen-disabled` attribute and `fullscreenDisabled` property for `dialog` across resolutions with `5.0.0-next.80`:

```html
<calcite-dialog open modal heading="Add a group" id="example-dialog" fullscreen-disabled>
 ...
</calcite-dialog>
```

<img width="330" height="342" alt="Image" src="https://github.com/user-attachments/assets/bcfd9397-6352-4682-aea0-de8cc75e1180" />
